### DISCUSSION TITLE
Defining retention period for partitions at tenant level

### DISCUSSION THREAD

**Discussion by anandsowm** [citusdata/citus#5647] [Q&A]
For a multi tenant application, that uses Citus and pg_partman, I want to be able to set the retention time for the partitions based on each Tenant's config setting. As per the docs, the retention is specified using a query like the one below, this means the retention is set for 1 month for all the tenants. How do I set the retention for each tenant? I think this requirement is quite common and hence asking it here, though the ability really need to be present in 'partman' extension.

`UPDATE partman.part_config SET retention_keep_table = false, retention = '1 month' 
WHERE parent_table = 'github.events';`

**Comment by marcocitus** [comment]
When sharding by tenant ID and partitioning by time, many tenants share the same partitioned table. The partiitioning hierarchy is global, so there is no way to drop partitions only for specific tenants or even specific shards. The workaround right now would be to use DELETE.

**Reply by anandsowm** [reply]
Thanks for the response. Deleting lot of rows from partitions would be inefficient and time consuming, but seems that is the currently available workaround. It would have been easier to detach a partition if somehow there was ability to distinguish '_distributed and partitioned_' tables and its partitions, so that retention can be specified based on each value for _distribution column_.
